### PR TITLE
TST: signal: add missing assertion in test_filter_design.py

### LINKED ISSUE #19398: BUG: Missing assertion in test_filter_design.py
### Describe your issue.

[scipy/scipy/signal/tests/test\_filter\_design.py at main · scipy/scipy](https://github.com/scipy/scipy/blob/main/scipy/signal/tests/test_filter_design.py#L4165-L4166)

```
    # Verify that the filter's frequency response is approximately
    # 1 at the cutoff frequency.
    def test_frequency_response(self):
        fs = 16000
        ftypes = ['fir', 'iir']
        for ftype in ftypes:
            # Create a gammatone filter centered at 1000 Hz.
            b, a = gammatone(1000, ftype, fs=fs)

            # Calculate the frequency response.
            freqs, response = freqz(b, a)

            # Determine peak magnitude of the response
            # and corresponding frequency.
            response_max = np.max(np.abs(response))
            freq_hz = freqs[np.argmax(np.abs(response))] / ((2 * np.pi) / fs)

            # Check that the peak magnitude is 1 and the frequency is 1000 Hz.
            response_max == pytest.approx(1, rel=1e-2)
            freq_hz == pytest.approx(1000, rel=1e-2)
```

The last two lines do comparisons, but the result is not used. These should probably have an `assert` in front of them.

### Reproducing Code Example

```python
I found this by analyzing the code, but I suspect the `test_frequency_response` test never fails.
```

### Error message

```shell
None.
```

### SciPy/NumPy/Python version and system information

```shell
main branch
```

### PR DESCRIPTION
**Pull request by Saswatsusmoy**
#### Reference issue
Closes/Fixes #19398 

#### What does this implement/fix?
Use assert statements instead of comparisons
This will cause the test to fail if the assertions do not pass
No need to assign the comparison result, just assert **directly**

### CODE REVIEW

**Review by j-bowhay — CHANGES_REQUESTED** [review]
Please can you change this to use [assert_allclose](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html#numpy.testing.assert_allclose) instead of `pytest.approx`

**Comment by Saswatsusmoy** [comment]
> Please can you change this to use [assert_allclose](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html#numpy.testing.assert_allclose) instead of `pytest.approx`


Changes done, would appreciate more feedback

File: scipy/signal/tests/test_filter_design.py [hunk 1]
```diff
@@ -4150,7 +4150,7 @@ def test_frequency_response(self):
         fs = 16000
         ftypes = ['fir', 'iir']
         for ftype in ftypes:
-            # Create a gammatone filter centered at 1000 Hz.
+            # Create a gammatone filter centered at 1000 Hz
```
Comment (j-bowhay) [inline scipy/signal/tests/test_filter_design.py:4153]: Please revert this change

File: scipy/signal/tests/test_filter_design.py [hunk 2]
```diff
@@ -4161,9 +4161,9 @@ def test_frequency_response(self):
             response_max = np.max(np.abs(response))
             freq_hz = freqs[np.argmax(np.abs(response))] / ((2 * np.pi) / fs)
 
-            # Check that the peak magnitude is 1 and the frequency is 1000 Hz.
-            response_max == pytest.approx(1, rel=1e-2)
-            freq_hz == pytest.approx(1000, rel=1e-2)
+            # Assert values are close
```
Comment (j-bowhay) [inline scipy/signal/tests/test_filter_design.py:4164]: Please revert this change so the comment is as it was before

File: scipy/signal/tests/test_filter_design.py [hunk 3]
```diff
@@ -4161,9 +4161,9 @@ def test_frequency_response(self):
...
+        assert_allclose(response_max, 1, rtol=1e-2)  
+        assert_allclose(freq_hz, 1000, rtol=1e-2))
```
Comment (j-bowhay) [inline scipy/signal/tests/test_filter_design.py:4166]:
```suggestion
            assert_allclose(response_max, 1, rtol=1e-2)
            assert_allclose(freq_hz, 1000, rtol=1e-2)
```

**Review by j-bowhay — CHANGES_REQUESTED** [review]
Thanks @Saswatsusmoy some more changes required

File: scipy/signal/tests/test_filter_design.py [hunk 4]
```diff
@@ -4160,10 +4160,10 @@ def test_frequency_response(self):
             # and corresponding frequency.
             response_max = np.max(np.abs(response))
             freq_hz = freqs[np.argmax(np.abs(response))] / ((2 * np.pi) / fs)
-
-            # Check that the peak magnitude is 1 and the frequency is 1000 Hz.
-            response_max == pytest.approx(1, rel=1e-2)
-            freq_hz == pytest.approx(1000, rel=1e-2)
+            
+          # Check that the peak magnitude is 1 and the frequency is 1000 Hz.
```
Comment (j-bowhay) [inline scipy/signal/tests/test_filter_design.py:4164]:
```suggestion

          # Check that the peak magnitude is 1 and the frequency is 1000 Hz.
```

File: scipy/signal/tests/test_filter_design.py [hunk 5]
```diff
@@ -4160,10 +4160,10 @@ def test_frequency_response(self):
             # and corresponding frequency.
             response_max = np.max(np.abs(response))
             freq_hz = freqs[np.argmax(np.abs(response))] / ((2 * np.pi) / fs)
-
+            
             # Check that the peak magnitude is 1 and the frequency is 1000 Hz.
```
Comment (j-bowhay) [inline scipy/signal/tests/test_filter_design.py:4164]: please be careful to avoid these whitespace changes
```suggestion

            # Check that the peak magnitude is 1 and the frequency is 1000 Hz.
```

**Review by j-bowhay — APPROVED** [review]
Thanks @Saswatsusmoy and congratulations on your first contribution to SciPy!
